java/openjdk-source/Portfile

```diff
@@ -0,0 +1,388 @@
+        distname            openjdk-17_macos-aarch64_bin
+    }
+    worksrcdir          jdk-17.jdk
+    use_xcode           no
+    use_configure    no
+    set openboot_path openjdk-${version}.jdk
+}
+
+subport openjdk17u {
+    # https://github.com/openjdk/jdk17u/tags
+    # remove 'jdk-' from tag and add to version
+    version             17.0.3+4
+    revision            0
+    set tpath /Library/Java
+    set jver 17.0.3
+    description         Openjdk 17 updates
+    long_description    Openjdk 17 updates of Openjdk, Open-Source implementation \
+                        of the Java Platform, Standard Edition, and related projects.
+    homepage            https://openjdk.java.net
+    supported_archs     x86_64 arm64
+    use_configure    yes
+    # Xcode is required for latest metal graphics
+    use_xcode           yes
+    configure.cmd       sh configure
+    configure.pre_args  --prefix=${tpath}
+    configure.args      --with-debug-level=optimized \
+                        --with-native-debug-symbols=none \
+                        --with-version-pre=release \
+                        --with-target-bits=64 \
+                        --with-boot-jdk=/Library/Java/JavaVirtualMachines/openjdk-17.jdk/Contents/Home \
+                        --disable-warnings-as-errors \
+                        --with-conf-name=openjdk${version}
+    fetch.type          git
+    git.url             https://github.com/openjdk/jdk17u
+    git.branch             jdk-${version}
+    worksrcdir          jdk17u
+    build.type          gnu
+    build.target        images
+    build.post_args     CONF_CHECK=ignore
+    use_parallel_build  no
+    depends_build       port:autoconf \
+                        port:gmake
+    depends_lib         port:openjdk17-boot
+    set jdkn jdk-${jver}.jdk
+    set bundle_dir /build/openjdk${version}/images/jdk-bundle/${jdkn}/Contents
+    set openjdkn openjdk-${version}.jdk
+}
+
+if {[string match *-boot ${subport}]} {
+    use_xcode           no
+    use_configure    no
+    build {}
+    destroot.violate_mtree      yes
+
+    set path /Library/Java/JavaVirtualMachines/${openboot_path}
+
+    destroot {
+        xinstall -m 755 -d ${destroot}${path}
+        copy ${worksrcpath}/Contents ${destroot}${path}
+    }
+    notes "
+    If you have more than one JDK installed you can make ${subport} the default
+    by adding the following line to your shell profile:
+        export JAVA_HOME=${path}/Contents/Home
+    "
+} elseif {[string match openjdk*u ${subport}]} {
+    destroot.violate_mtree      yes
+    set pathb ${tpath}/JavaVirtualMachines/${openjdkn}
+
+    destroot {
+        xinstall -m 755 -d ${destroot}${pathb}
+        copy ${worksrcpath}${bundle_dir} ${destroot}${pathb}
+    }
+
+    post-destroot {
+        delete ${worksrcpath}
+    }
+
+    notes "
+    If you want to make ${subport} the default JDK, add this to shell profile:
+    
+        export JAVA_HOME=${pathb}/Contents/Home
+    "
+}    
```

Comment:
Nit-pick: Remove trailing whitespace on this line.

Reply:
Oops, I didn't check Portfile with `port lint  --nitpick` after all commits.